common/presentation/src/main/kotlin/com/pandulapeter/myTodos/presentation/ui/Demo2.kt

```diff
@@ -18,80 +20,66 @@ import com.pandulapeter.myTodos.model.Issue
 import com.pandulapeter.myTodos.presentation.resources.Dimension
 import com.pandulapeter.myTodos.presentation.resources.Text
 import com.pandulapeter.myTodos.presentation.utilities.get
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
 
 @Composable
-internal fun ColumnScope.Demo2(
-    getIssues: GetIssuesUseCase = get()
+internal fun Demo2(
+    getIssues: GetIssuesUseCase = get(),
 ) {
     var issues by remember { mutableStateOf<ResultWrapper<List<Issue>>?>(null) }
+    var retry by remember { mutableStateOf(Any()) }
 
-    fun loadIssues() {
-        issues = null
-        GlobalScope.launch {
-            val newIssues = getIssues()
-            launch(Dispatchers.Main) {
-                issues = newIssues
-            }
-        }
+    LaunchedEffect(getIssues, retry) {
+        issues = getIssues()
     }
 
-    when (val currentIssues = issues) {
-        null -> {
-            LoadingIndicator()
-            loadIssues()
-        }
-        is ResultWrapper.Error -> {
-            ErrorState(
-                onTryAgainButtonPressed = ::loadIssues
+    Column(
+        modifier = Modifier.fillMaxWidth(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        when (val currentIssues = issues) {
+            null -> LoadingIndicator()
+            is ResultWrapper.Error -> ErrorState(
+                onTryAgainButtonPressed = { issues = null; retry = Any() }
             )
-        }
-        is ResultWrapper.Success -> {
-            IdleState(
+            is ResultWrapper.Success -> IdleState(
                 issues = currentIssues.data
             )
         }
     }
 }
 
 @Composable
-private fun ColumnScope.LoadingIndicator() = CircularProgressIndicator(
+private fun LoadingIndicator(
+    modifier: Modifier = Modifier,
```

Comment:
I noticed the pattern for UI components in Compose is to have a Modifier as the first parameter to let the parent apply some modifiers which are not child specific. I would argue that in this case this padding is not specific to this loading indicator, but instead is something that the parent needs.